### ISSUE TITLE
SYCL mdrange_parallel_reduce_primitive_types unit test spurious failures on Ponte Vecchio

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#6501]
**Describe the bug**
The `sycl.mdrange_parallel_reduce_primitive_types` has failed spuriously in builds on Ponte Vecchio (not due to recent changes). This is the output for reference:

```
[ RUN      ] sycl.mdrange_parallel_reduce_primitive_types
<https://jenkins-son.sandia.gov/job/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/ws/kokkos/core/unit_test/TestMDRangeReduce.hpp>:44: Failure
Expected equality of these values:
  land_MD
    Which is: 0
  1
<https://jenkins-son.sandia.gov/job/Kokkos_develop_blake_sycl_pv_oneapi_2023_1_0/ws/kokkos/core/unit_test/TestMDRangeReduce.hpp>:44: Failure
Expected equality of these values:
  land_MD
    Which is: 0
  1
[  FAILED  ] sycl.mdrange_parallel_reduce_primitive_types (8060 ms)
```

This has also been discussed here for example https://github.com/kokkos/kokkos/pull/6430#issuecomment-1714686875 , and presumably #6430 may resolve this

**Please also include the following items to support reproducing the bug**
**Reproducer** (Blake PV queue):
```bash
module load cmake intel-oneapi-compilers/2023.1.0

$KOKKOS_PATH/generate_makefile.bash --with-sycl --arch=INTEL_PVC --compiler=/projects/x86-64-icelake-rocky8/compilers/intel-oneapi-compilers/2023.1.0/gcc/8.5.0/base/6g2jkiv/compiler/2023.1.0/linux/bin-llvm/clang++ --cxxflags="-fp-model=precise" --cmake-flags=-DKokkos_ENABLE_ONEDPL=OFF
```

**Comment by masterleinad** [comment]
Can you try if #6430 fixes this indeed?

**Comment by ndellingwood** [comment]
@masterleinad that PR seems to fix the failing test, no fails after 10 repeats (usually fails at least 50% of the time or so)

**Comment by masterleinad** [comment]
Fixed in #6430.
